client-app/shared/cart/composables/useCartExtensionPoints.ts

```diff
@@ -0,0 +1,26 @@
+import { createGlobalState } from "@vueuse/core";
+import { ref } from "vue";
+import type { DefineComponent } from "vue";
+
+type ElementType = {
+  id: string;
+  /* eslint-disable @typescript-eslint/no-explicit-any */
+  element: DefineComponent<Record<string, any>, Record<string, any>, any>;
+};
+
+function _useCartExtensionPoints() {
+  const sidebarWidgets = ref<ElementType[]>([]);
+
+  function registerSidebarWidget(element: ElementType) {
+    if (!sidebarWidgets.value.find((el) => el.id === element.id)) {
```

Comment:
better use `some` instead of `find` as we don't need an element here, just checking if the array contains that